addon/bootstrap.js

```diff
@@ -105,8 +108,8 @@ async function startup(addonData, reason) {
   // log what the study variation and other info is.
   log.debug(`info ${JSON.stringify(studyUtils.info())}`);
 
-  // Start up your feature, with specific variation info.
-  this.feature = new Feature({variation, studyUtils, reasonName: REASONS[reason]});
+  // start up the chrome-privileged part of the study
+  this.feature.privilegedStartup();
```

Comment:
Just curious, why is this better than instantiating the feature once and having its constructor method startup the feature?

Reply:
Originally, I separated them for the ability to have to extension startup hooks - one for when the webextension had loaded asynchronously, and one at the end of startup() for privileged startup logic. Currently, there is only one startup method but it is anyway clearer to not have side-effects happen in a constructor event.